{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import time \n",
    "import mxnet as mx \n",
    "from mxnet import gluon, autograd\n",
    "from data_loader import videoFolder\n",
    "import utils\n",
    "from option import Options, args_\n",
    "from multiprocessing import cpu_count\n",
    "from network import lstm_net, L2Loss_2, L2Loss_cos\n",
    "import sys"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def train(args):\n",
    "    frames = args.frames\n",
    "    caption_length = args.caption_length\n",
    "    glove_file = args.glove_file\n",
    "\n",
    "    CPU_COUNT = cpu_count()\n",
    "\n",
    "    if args.cuda:\n",
    "        ctx = mx.gpu()\n",
    "    else:\n",
    "        ctx = mx.cpu()\n",
    "    \n",
    "    transform = utils.Compose([utils.ToTensor(ctx),\n",
    "                               utils.normalize(ctx),\n",
    "                               ])\n",
    "\n",
    "    train_dataset = videoFolder(args.train_folder,args.train_dict, frames, glove_file, caption_length, transform=transform)\n",
    "\n",
    "    #test_dataset = videoFolder(args.test_folder,args.test_dict, frames, glove_file, caption_length, transform=transform)\n",
    "\n",
    "    train_loader = gluon.data.DataLoader(train_dataset,batch_size=args.batch_size,last_batch='keep',shuffle=True, num_workers=CPU_COUNT)\n",
    "\n",
    "    #test_loader = gluon.data.DataLoader(test_dataset,batch_size=args.batch_size,last_batch='keep',shuffle=False,num_workers=CPU_COUNT)\n",
    "\n",
    "    loss = L2Loss_cos(ctx)\n",
    "    net = lstm_net(frames,caption_length)\n",
    "    net.initialize(init=mx.initializer.MSRAPrelu(), ctx=ctx)\n",
    "    trainer = gluon.Trainer(net.collect_params(), 'adam',\n",
    "                            {'learning_rate': args.lr})\n",
    "\n",
    "    for e in range(args.epochs):\n",
    "        for batch_id, (x,_) in enumerate(train_loader):\n",
    "            with autograd.record():\n",
    "                pred = net(x)\n",
    "                train_loss = loss(pred,_)\n",
    "                train_loss.backward()\n",
    "            \n",
    "            print(train_loss.shape)\n",
    "            \n",
    "            trainer.step(args.batch_size)\n",
    "            mx.nd.waitall()\n",
    "        \n",
    "        #print(\"Epoch {}, train_loss: {:.4f}\".format(e+1, train_loss))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def main():\n",
    "    args = args_()\n",
    "    train(args)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Process Process-4:\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/usr/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/gluon/data/dataloader.py\", line 184, in worker_loop\n",
      "    batch = batchify_fn([dataset[i] for i in samples])\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/gluon/data/dataloader.py\", line 184, in <listcomp>\n",
      "    batch = batchify_fn([dataset[i] for i in samples])\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/data_loader.py\", line 104, in __getitem__\n",
      "    video_frame = self.transform(video_frame)\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/utils.py\", line 128, in __call__\n",
      "    img = t(img)\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/utils.py\", line 93, in __call__\n",
      "    img = mx.nd.array(np.array(img).transpose(0, 3, 1, 2).astype('float32'), ctx=self.ctx)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/utils.py\", line 146, in array\n",
      "    return _array(source_array, ctx=ctx, dtype=dtype)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 2434, in array\n",
      "    arr = empty(source_array.shape, ctx, dtype)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 3823, in empty\n",
      "    return NDArray(handle=_new_alloc_handle(shape, ctx, False, dtype))\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 139, in _new_alloc_handle\n",
      "    ctypes.byref(hdl)))\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/base.py\", line 251, in check_call\n",
      "    raise MXNetError(py_str(_LIB.MXGetLastError()))\n",
      "mxnet.base.MXNetError: [22:12:03] src/storage/storage.cc:65: Check failed: e == cudaSuccess || e == cudaErrorCudartUnloading CUDA: initialization error\n",
      "\n",
      "Stack trace returned 10 entries:\n",
      "[bt] (0) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x39008a) [0x7fe75a3ed08a]\n",
      "[bt] (1) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x3906c1) [0x7fe75a3ed6c1]\n",
      "[bt] (2) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x31aba9d) [0x7fe75d208a9d]\n",
      "[bt] (3) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x31b3e58) [0x7fe75d210e58]\n",
      "[bt] (4) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x2a808f6) [0x7fe75cadd8f6]\n",
      "[bt] (5) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(MXNDArrayCreateEx+0x158) [0x7fe75cade168]\n",
      "[bt] (6) /usr/lib/x86_64-linux-gnu/libffi.so.6(ffi_call_unix64+0x4c) [0x7fe7bee55dae]\n",
      "[bt] (7) /usr/lib/x86_64-linux-gnu/libffi.so.6(ffi_call+0x22f) [0x7fe7bee5571f]\n",
      "[bt] (8) /usr/lib/python3.6/lib-dynload/_ctypes.cpython-36m-x86_64-linux-gnu.so(_ctypes_callproc+0x2b4) [0x7fe7bf069524]\n",
      "[bt] (9) /usr/lib/python3.6/lib-dynload/_ctypes.cpython-36m-x86_64-linux-gnu.so(+0x11b93) [0x7fe7bf069b93]\n",
      "\n",
      "\n",
      "Process Process-1:\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/usr/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/gluon/data/dataloader.py\", line 184, in worker_loop\n",
      "    batch = batchify_fn([dataset[i] for i in samples])\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/gluon/data/dataloader.py\", line 184, in <listcomp>\n",
      "    batch = batchify_fn([dataset[i] for i in samples])\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/data_loader.py\", line 104, in __getitem__\n",
      "    video_frame = self.transform(video_frame)\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/utils.py\", line 128, in __call__\n",
      "    img = t(img)\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/utils.py\", line 93, in __call__\n",
      "    img = mx.nd.array(np.array(img).transpose(0, 3, 1, 2).astype('float32'), ctx=self.ctx)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/utils.py\", line 146, in array\n",
      "    return _array(source_array, ctx=ctx, dtype=dtype)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 2434, in array\n",
      "    arr = empty(source_array.shape, ctx, dtype)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 3823, in empty\n",
      "    return NDArray(handle=_new_alloc_handle(shape, ctx, False, dtype))\n",
      "Process Process-3:\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 139, in _new_alloc_handle\n",
      "    ctypes.byref(hdl)))\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/base.py\", line 251, in check_call\n",
      "    raise MXNetError(py_str(_LIB.MXGetLastError()))\n",
      "mxnet.base.MXNetError: [22:12:03] src/storage/storage.cc:65: Check failed: e == cudaSuccess || e == cudaErrorCudartUnloading CUDA: initialization error\n",
      "\n",
      "Stack trace returned 10 entries:\n",
      "[bt] (0) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x39008a) [0x7fe75a3ed08a]\n",
      "[bt] (1) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x3906c1) [0x7fe75a3ed6c1]\n",
      "[bt] (2) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x31aba9d) [0x7fe75d208a9d]\n",
      "[bt] (3) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x31b3e58) [0x7fe75d210e58]\n",
      "[bt] (4) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x2a808f6) [0x7fe75cadd8f6]\n",
      "[bt] (5) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(MXNDArrayCreateEx+0x158) [0x7fe75cade168]\n",
      "[bt] (6) /usr/lib/x86_64-linux-gnu/libffi.so.6(ffi_call_unix64+0x4c) [0x7fe7bee55dae]\n",
      "[bt] (7) /usr/lib/x86_64-linux-gnu/libffi.so.6(ffi_call+0x22f) [0x7fe7bee5571f]\n",
      "[bt] (8) /usr/lib/python3.6/lib-dynload/_ctypes.cpython-36m-x86_64-linux-gnu.so(_ctypes_callproc+0x2b4) [0x7fe7bf069524]\n",
      "[bt] (9) /usr/lib/python3.6/lib-dynload/_ctypes.cpython-36m-x86_64-linux-gnu.so(+0x11b93) [0x7fe7bf069b93]\n",
      "\n",
      "\n",
      "  File \"/usr/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/usr/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/gluon/data/dataloader.py\", line 184, in worker_loop\n",
      "    batch = batchify_fn([dataset[i] for i in samples])\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/gluon/data/dataloader.py\", line 184, in <listcomp>\n",
      "    batch = batchify_fn([dataset[i] for i in samples])\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/data_loader.py\", line 104, in __getitem__\n",
      "    video_frame = self.transform(video_frame)\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/utils.py\", line 128, in __call__\n",
      "    img = t(img)\n",
      "Process Process-6:\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/utils.py\", line 93, in __call__\n",
      "    img = mx.nd.array(np.array(img).transpose(0, 3, 1, 2).astype('float32'), ctx=self.ctx)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/utils.py\", line 146, in array\n",
      "    return _array(source_array, ctx=ctx, dtype=dtype)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 2434, in array\n",
      "    arr = empty(source_array.shape, ctx, dtype)\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 3823, in empty\n",
      "    return NDArray(handle=_new_alloc_handle(shape, ctx, False, dtype))\n",
      "  File \"/usr/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 139, in _new_alloc_handle\n",
      "    ctypes.byref(hdl)))\n",
      "  File \"/usr/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/base.py\", line 251, in check_call\n",
      "    raise MXNetError(py_str(_LIB.MXGetLastError()))\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/gluon/data/dataloader.py\", line 184, in worker_loop\n",
      "    batch = batchify_fn([dataset[i] for i in samples])\n",
      "mxnet.base.MXNetError: [22:12:03] src/storage/storage.cc:65: Check failed: e == cudaSuccess || e == cudaErrorCudartUnloading CUDA: initialization error\n",
      "\n",
      "Stack trace returned 10 entries:\n",
      "[bt] (0) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x39008a) [0x7fe75a3ed08a]\n",
      "[bt] (1) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x3906c1) [0x7fe75a3ed6c1]\n",
      "[bt] (2) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x31aba9d) [0x7fe75d208a9d]\n",
      "[bt] (3) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x31b3e58) [0x7fe75d210e58]\n",
      "[bt] (4) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x2a808f6) [0x7fe75cadd8f6]\n",
      "[bt] (5) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(MXNDArrayCreateEx+0x158) [0x7fe75cade168]\n",
      "[bt] (6) /usr/lib/x86_64-linux-gnu/libffi.so.6(ffi_call_unix64+0x4c) [0x7fe7bee55dae]\n",
      "[bt] (7) /usr/lib/x86_64-linux-gnu/libffi.so.6(ffi_call+0x22f) [0x7fe7bee5571f]\n",
      "[bt] (8) /usr/lib/python3.6/lib-dynload/_ctypes.cpython-36m-x86_64-linux-gnu.so(_ctypes_callproc+0x2b4) [0x7fe7bf069524]\n",
      "[bt] (9) /usr/lib/python3.6/lib-dynload/_ctypes.cpython-36m-x86_64-linux-gnu.so(+0x11b93) [0x7fe7bf069b93]\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/gluon/data/dataloader.py\", line 184, in <listcomp>\n",
      "    batch = batchify_fn([dataset[i] for i in samples])\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/data_loader.py\", line 104, in __getitem__\n",
      "    video_frame = self.transform(video_frame)\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/utils.py\", line 128, in __call__\n",
      "    img = t(img)\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/utils.py\", line 93, in __call__\n",
      "    img = mx.nd.array(np.array(img).transpose(0, 3, 1, 2).astype('float32'), ctx=self.ctx)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/utils.py\", line 146, in array\n",
      "    return _array(source_array, ctx=ctx, dtype=dtype)\n",
      "Process Process-9:\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 2434, in array\n",
      "    arr = empty(source_array.shape, ctx, dtype)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 3823, in empty\n",
      "    return NDArray(handle=_new_alloc_handle(shape, ctx, False, dtype))\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 139, in _new_alloc_handle\n",
      "    ctypes.byref(hdl)))\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/base.py\", line 251, in check_call\n",
      "    raise MXNetError(py_str(_LIB.MXGetLastError()))\n",
      "  File \"/usr/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/usr/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "mxnet.base.MXNetError: [22:12:03] src/storage/storage.cc:65: Check failed: e == cudaSuccess || e == cudaErrorCudartUnloading CUDA: initialization error\n",
      "\n",
      "Stack trace returned 10 entries:\n",
      "[bt] (0) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x39008a) [0x7fe75a3ed08a]\n",
      "[bt] (1) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x3906c1) [0x7fe75a3ed6c1]\n",
      "[bt] (2) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x31aba9d) [0x7fe75d208a9d]\n",
      "[bt] (3) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x31b3e58) [0x7fe75d210e58]\n",
      "[bt] (4) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x2a808f6) [0x7fe75cadd8f6]\n",
      "[bt] (5) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(MXNDArrayCreateEx+0x158) [0x7fe75cade168]\n",
      "[bt] (6) /usr/lib/x86_64-linux-gnu/libffi.so.6(ffi_call_unix64+0x4c) [0x7fe7bee55dae]\n",
      "[bt] (7) /usr/lib/x86_64-linux-gnu/libffi.so.6(ffi_call+0x22f) [0x7fe7bee5571f]\n",
      "[bt] (8) /usr/lib/python3.6/lib-dynload/_ctypes.cpython-36m-x86_64-linux-gnu.so(_ctypes_callproc+0x2b4) [0x7fe7bf069524]\n",
      "[bt] (9) /usr/lib/python3.6/lib-dynload/_ctypes.cpython-36m-x86_64-linux-gnu.so(+0x11b93) [0x7fe7bf069b93]\n",
      "\n",
      "\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/gluon/data/dataloader.py\", line 184, in worker_loop\n",
      "    batch = batchify_fn([dataset[i] for i in samples])\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/gluon/data/dataloader.py\", line 184, in <listcomp>\n",
      "    batch = batchify_fn([dataset[i] for i in samples])\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/data_loader.py\", line 104, in __getitem__\n",
      "    video_frame = self.transform(video_frame)\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/utils.py\", line 128, in __call__\n",
      "    img = t(img)\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/utils.py\", line 93, in __call__\n",
      "    img = mx.nd.array(np.array(img).transpose(0, 3, 1, 2).astype('float32'), ctx=self.ctx)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/utils.py\", line 146, in array\n",
      "    return _array(source_array, ctx=ctx, dtype=dtype)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 2434, in array\n",
      "    arr = empty(source_array.shape, ctx, dtype)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 3823, in empty\n",
      "    return NDArray(handle=_new_alloc_handle(shape, ctx, False, dtype))\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 139, in _new_alloc_handle\n",
      "    ctypes.byref(hdl)))\n",
      "Process Process-2:\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/base.py\", line 251, in check_call\n",
      "    raise MXNetError(py_str(_LIB.MXGetLastError()))\n",
      "mxnet.base.MXNetError: [22:12:03] src/storage/storage.cc:65: Check failed: e == cudaSuccess || e == cudaErrorCudartUnloading CUDA: initialization error\n",
      "\n",
      "Stack trace returned 10 entries:\n",
      "[bt] (0) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x39008a) [0x7fe75a3ed08a]\n",
      "[bt] (1) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x3906c1) [0x7fe75a3ed6c1]\n",
      "[bt] (2) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x31aba9d) [0x7fe75d208a9d]\n",
      "[bt] (3) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x31b3e58) [0x7fe75d210e58]\n",
      "[bt] (4) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x2a808f6) [0x7fe75cadd8f6]\n",
      "[bt] (5) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(MXNDArrayCreateEx+0x158) [0x7fe75cade168]\n",
      "[bt] (6) /usr/lib/x86_64-linux-gnu/libffi.so.6(ffi_call_unix64+0x4c) [0x7fe7bee55dae]\n",
      "[bt] (7) /usr/lib/x86_64-linux-gnu/libffi.so.6(ffi_call+0x22f) [0x7fe7bee5571f]\n",
      "[bt] (8) /usr/lib/python3.6/lib-dynload/_ctypes.cpython-36m-x86_64-linux-gnu.so(_ctypes_callproc+0x2b4) [0x7fe7bf069524]\n",
      "[bt] (9) /usr/lib/python3.6/lib-dynload/_ctypes.cpython-36m-x86_64-linux-gnu.so(+0x11b93) [0x7fe7bf069b93]\n",
      "\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/usr/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/gluon/data/dataloader.py\", line 184, in worker_loop\n",
      "    batch = batchify_fn([dataset[i] for i in samples])\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/gluon/data/dataloader.py\", line 184, in <listcomp>\n",
      "    batch = batchify_fn([dataset[i] for i in samples])\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/data_loader.py\", line 104, in __getitem__\n",
      "    video_frame = self.transform(video_frame)\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/utils.py\", line 128, in __call__\n",
      "    img = t(img)\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/utils.py\", line 93, in __call__\n",
      "    img = mx.nd.array(np.array(img).transpose(0, 3, 1, 2).astype('float32'), ctx=self.ctx)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/utils.py\", line 146, in array\n",
      "    return _array(source_array, ctx=ctx, dtype=dtype)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 2434, in array\n",
      "    arr = empty(source_array.shape, ctx, dtype)\n",
      "Process Process-10:\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 3823, in empty\n",
      "    return NDArray(handle=_new_alloc_handle(shape, ctx, False, dtype))\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 139, in _new_alloc_handle\n",
      "    ctypes.byref(hdl)))\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/base.py\", line 251, in check_call\n",
      "    raise MXNetError(py_str(_LIB.MXGetLastError()))\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/usr/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "mxnet.base.MXNetError: [22:12:03] src/storage/storage.cc:65: Check failed: e == cudaSuccess || e == cudaErrorCudartUnloading CUDA: initialization error\n",
      "\n",
      "Stack trace returned 10 entries:\n",
      "[bt] (0) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x39008a) [0x7fe75a3ed08a]\n",
      "[bt] (1) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x3906c1) [0x7fe75a3ed6c1]\n",
      "[bt] (2) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x31aba9d) [0x7fe75d208a9d]\n",
      "[bt] (3) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x31b3e58) [0x7fe75d210e58]\n",
      "[bt] (4) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x2a808f6) [0x7fe75cadd8f6]\n",
      "[bt] (5) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(MXNDArrayCreateEx+0x158) [0x7fe75cade168]\n",
      "[bt] (6) /usr/lib/x86_64-linux-gnu/libffi.so.6(ffi_call_unix64+0x4c) [0x7fe7bee55dae]\n",
      "[bt] (7) /usr/lib/x86_64-linux-gnu/libffi.so.6(ffi_call+0x22f) [0x7fe7bee5571f]\n",
      "[bt] (8) /usr/lib/python3.6/lib-dynload/_ctypes.cpython-36m-x86_64-linux-gnu.so(_ctypes_callproc+0x2b4) [0x7fe7bf069524]\n",
      "[bt] (9) /usr/lib/python3.6/lib-dynload/_ctypes.cpython-36m-x86_64-linux-gnu.so(+0x11b93) [0x7fe7bf069b93]\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/gluon/data/dataloader.py\", line 184, in worker_loop\n",
      "    batch = batchify_fn([dataset[i] for i in samples])\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/gluon/data/dataloader.py\", line 184, in <listcomp>\n",
      "    batch = batchify_fn([dataset[i] for i in samples])\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/data_loader.py\", line 104, in __getitem__\n",
      "    video_frame = self.transform(video_frame)\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/utils.py\", line 128, in __call__\n",
      "    img = t(img)\n",
      "Process Process-12:\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/utils.py\", line 93, in __call__\n",
      "    img = mx.nd.array(np.array(img).transpose(0, 3, 1, 2).astype('float32'), ctx=self.ctx)\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/utils.py\", line 146, in array\n",
      "    return _array(source_array, ctx=ctx, dtype=dtype)\n",
      "  File \"/usr/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/usr/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/gluon/data/dataloader.py\", line 184, in worker_loop\n",
      "    batch = batchify_fn([dataset[i] for i in samples])\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/gluon/data/dataloader.py\", line 184, in <listcomp>\n",
      "    batch = batchify_fn([dataset[i] for i in samples])\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/data_loader.py\", line 104, in __getitem__\n",
      "    video_frame = self.transform(video_frame)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 2434, in array\n",
      "    arr = empty(source_array.shape, ctx, dtype)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 3823, in empty\n",
      "    return NDArray(handle=_new_alloc_handle(shape, ctx, False, dtype))\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/utils.py\", line 128, in __call__\n",
      "    img = t(img)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 139, in _new_alloc_handle\n",
      "    ctypes.byref(hdl)))\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/utils.py\", line 93, in __call__\n",
      "    img = mx.nd.array(np.array(img).transpose(0, 3, 1, 2).astype('float32'), ctx=self.ctx)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/base.py\", line 251, in check_call\n",
      "    raise MXNetError(py_str(_LIB.MXGetLastError()))\n",
      "mxnet.base.MXNetError: [22:12:03] src/storage/storage.cc:65: Check failed: e == cudaSuccess || e == cudaErrorCudartUnloading CUDA: initialization error\n",
      "\n",
      "Stack trace returned 10 entries:\n",
      "[bt] (0) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x39008a) [0x7fe75a3ed08a]\n",
      "[bt] (1) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x3906c1) [0x7fe75a3ed6c1]\n",
      "[bt] (2) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x31aba9d) [0x7fe75d208a9d]\n",
      "[bt] (3) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x31b3e58) [0x7fe75d210e58]\n",
      "[bt] (4) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x2a808f6) [0x7fe75cadd8f6]\n",
      "[bt] (5) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(MXNDArrayCreateEx+0x158) [0x7fe75cade168]\n",
      "[bt] (6) /usr/lib/x86_64-linux-gnu/libffi.so.6(ffi_call_unix64+0x4c) [0x7fe7bee55dae]\n",
      "[bt] (7) /usr/lib/x86_64-linux-gnu/libffi.so.6(ffi_call+0x22f) [0x7fe7bee5571f]\n",
      "[bt] (8) /usr/lib/python3.6/lib-dynload/_ctypes.cpython-36m-x86_64-linux-gnu.so(_ctypes_callproc+0x2b4) [0x7fe7bf069524]\n",
      "[bt] (9) /usr/lib/python3.6/lib-dynload/_ctypes.cpython-36m-x86_64-linux-gnu.so(+0x11b93) [0x7fe7bf069b93]\n",
      "\n",
      "\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/utils.py\", line 146, in array\n",
      "    return _array(source_array, ctx=ctx, dtype=dtype)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 2434, in array\n",
      "    arr = empty(source_array.shape, ctx, dtype)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 3823, in empty\n",
      "    return NDArray(handle=_new_alloc_handle(shape, ctx, False, dtype))\n",
      "Process Process-11:\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 139, in _new_alloc_handle\n",
      "    ctypes.byref(hdl)))\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/base.py\", line 251, in check_call\n",
      "    raise MXNetError(py_str(_LIB.MXGetLastError()))\n",
      "mxnet.base.MXNetError: [22:12:04] src/storage/storage.cc:65: Check failed: e == cudaSuccess || e == cudaErrorCudartUnloading CUDA: initialization error\n",
      "\n",
      "Stack trace returned 10 entries:\n",
      "[bt] (0) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x39008a) [0x7fe75a3ed08a]\n",
      "[bt] (1) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x3906c1) [0x7fe75a3ed6c1]\n",
      "[bt] (2) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x31aba9d) [0x7fe75d208a9d]\n",
      "[bt] (3) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x31b3e58) [0x7fe75d210e58]\n",
      "[bt] (4) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x2a808f6) [0x7fe75cadd8f6]\n",
      "[bt] (5) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(MXNDArrayCreateEx+0x158) [0x7fe75cade168]\n",
      "[bt] (6) /usr/lib/x86_64-linux-gnu/libffi.so.6(ffi_call_unix64+0x4c) [0x7fe7bee55dae]\n",
      "[bt] (7) /usr/lib/x86_64-linux-gnu/libffi.so.6(ffi_call+0x22f) [0x7fe7bee5571f]\n",
      "[bt] (8) /usr/lib/python3.6/lib-dynload/_ctypes.cpython-36m-x86_64-linux-gnu.so(_ctypes_callproc+0x2b4) [0x7fe7bf069524]\n",
      "[bt] (9) /usr/lib/python3.6/lib-dynload/_ctypes.cpython-36m-x86_64-linux-gnu.so(+0x11b93) [0x7fe7bf069b93]\n",
      "\n",
      "\n",
      "  File \"/usr/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/gluon/data/dataloader.py\", line 184, in worker_loop\n",
      "    batch = batchify_fn([dataset[i] for i in samples])\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/gluon/data/dataloader.py\", line 184, in <listcomp>\n",
      "    batch = batchify_fn([dataset[i] for i in samples])\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/data_loader.py\", line 104, in __getitem__\n",
      "    video_frame = self.transform(video_frame)\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/utils.py\", line 128, in __call__\n",
      "    img = t(img)\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/utils.py\", line 93, in __call__\n",
      "    img = mx.nd.array(np.array(img).transpose(0, 3, 1, 2).astype('float32'), ctx=self.ctx)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/utils.py\", line 146, in array\n",
      "    return _array(source_array, ctx=ctx, dtype=dtype)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 2434, in array\n",
      "    arr = empty(source_array.shape, ctx, dtype)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 3823, in empty\n",
      "    return NDArray(handle=_new_alloc_handle(shape, ctx, False, dtype))\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 139, in _new_alloc_handle\n",
      "    ctypes.byref(hdl)))\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/base.py\", line 251, in check_call\n",
      "    raise MXNetError(py_str(_LIB.MXGetLastError()))\n",
      "mxnet.base.MXNetError: [22:12:04] src/storage/storage.cc:65: Check failed: e == cudaSuccess || e == cudaErrorCudartUnloading CUDA: initialization error\n",
      "\n",
      "Stack trace returned 10 entries:\n",
      "[bt] (0) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x39008a) [0x7fe75a3ed08a]\n",
      "[bt] (1) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x3906c1) [0x7fe75a3ed6c1]\n",
      "[bt] (2) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x31aba9d) [0x7fe75d208a9d]\n",
      "[bt] (3) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x31b3e58) [0x7fe75d210e58]\n",
      "[bt] (4) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x2a808f6) [0x7fe75cadd8f6]\n",
      "[bt] (5) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(MXNDArrayCreateEx+0x158) [0x7fe75cade168]\n",
      "[bt] (6) /usr/lib/x86_64-linux-gnu/libffi.so.6(ffi_call_unix64+0x4c) [0x7fe7bee55dae]\n",
      "[bt] (7) /usr/lib/x86_64-linux-gnu/libffi.so.6(ffi_call+0x22f) [0x7fe7bee5571f]\n",
      "[bt] (8) /usr/lib/python3.6/lib-dynload/_ctypes.cpython-36m-x86_64-linux-gnu.so(_ctypes_callproc+0x2b4) [0x7fe7bf069524]\n",
      "[bt] (9) /usr/lib/python3.6/lib-dynload/_ctypes.cpython-36m-x86_64-linux-gnu.so(+0x11b93) [0x7fe7bf069b93]\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Process Process-8:\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/usr/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/gluon/data/dataloader.py\", line 184, in worker_loop\n",
      "    batch = batchify_fn([dataset[i] for i in samples])\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/gluon/data/dataloader.py\", line 184, in <listcomp>\n",
      "    batch = batchify_fn([dataset[i] for i in samples])\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/data_loader.py\", line 104, in __getitem__\n",
      "    video_frame = self.transform(video_frame)\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/utils.py\", line 128, in __call__\n",
      "    img = t(img)\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/utils.py\", line 93, in __call__\n",
      "    img = mx.nd.array(np.array(img).transpose(0, 3, 1, 2).astype('float32'), ctx=self.ctx)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/utils.py\", line 146, in array\n",
      "    return _array(source_array, ctx=ctx, dtype=dtype)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 2434, in array\n",
      "    arr = empty(source_array.shape, ctx, dtype)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 3823, in empty\n",
      "    return NDArray(handle=_new_alloc_handle(shape, ctx, False, dtype))\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 139, in _new_alloc_handle\n",
      "    ctypes.byref(hdl)))\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/base.py\", line 251, in check_call\n",
      "    raise MXNetError(py_str(_LIB.MXGetLastError()))\n",
      "Process Process-5:\n",
      "mxnet.base.MXNetError: [22:12:04] src/storage/storage.cc:65: Check failed: e == cudaSuccess || e == cudaErrorCudartUnloading CUDA: initialization error\n",
      "\n",
      "Stack trace returned 10 entries:\n",
      "[bt] (0) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x39008a) [0x7fe75a3ed08a]\n",
      "[bt] (1) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x3906c1) [0x7fe75a3ed6c1]\n",
      "[bt] (2) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x31aba9d) [0x7fe75d208a9d]\n",
      "[bt] (3) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x31b3e58) [0x7fe75d210e58]\n",
      "[bt] (4) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x2a808f6) [0x7fe75cadd8f6]\n",
      "[bt] (5) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(MXNDArrayCreateEx+0x158) [0x7fe75cade168]\n",
      "[bt] (6) /usr/lib/x86_64-linux-gnu/libffi.so.6(ffi_call_unix64+0x4c) [0x7fe7bee55dae]\n",
      "[bt] (7) /usr/lib/x86_64-linux-gnu/libffi.so.6(ffi_call+0x22f) [0x7fe7bee5571f]\n",
      "[bt] (8) /usr/lib/python3.6/lib-dynload/_ctypes.cpython-36m-x86_64-linux-gnu.so(_ctypes_callproc+0x2b4) [0x7fe7bf069524]\n",
      "[bt] (9) /usr/lib/python3.6/lib-dynload/_ctypes.cpython-36m-x86_64-linux-gnu.so(+0x11b93) [0x7fe7bf069b93]\n",
      "\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/usr/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/gluon/data/dataloader.py\", line 184, in worker_loop\n",
      "    batch = batchify_fn([dataset[i] for i in samples])\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/gluon/data/dataloader.py\", line 184, in <listcomp>\n",
      "    batch = batchify_fn([dataset[i] for i in samples])\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/data_loader.py\", line 104, in __getitem__\n",
      "    video_frame = self.transform(video_frame)\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/utils.py\", line 128, in __call__\n",
      "    img = t(img)\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/utils.py\", line 93, in __call__\n",
      "    img = mx.nd.array(np.array(img).transpose(0, 3, 1, 2).astype('float32'), ctx=self.ctx)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/utils.py\", line 146, in array\n",
      "    return _array(source_array, ctx=ctx, dtype=dtype)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 2434, in array\n",
      "    arr = empty(source_array.shape, ctx, dtype)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 3823, in empty\n",
      "    return NDArray(handle=_new_alloc_handle(shape, ctx, False, dtype))\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 139, in _new_alloc_handle\n",
      "    ctypes.byref(hdl)))\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/base.py\", line 251, in check_call\n",
      "    raise MXNetError(py_str(_LIB.MXGetLastError()))\n",
      "mxnet.base.MXNetError: [22:12:04] src/storage/storage.cc:65: Check failed: e == cudaSuccess || e == cudaErrorCudartUnloading CUDA: initialization error\n",
      "\n",
      "Stack trace returned 10 entries:\n",
      "[bt] (0) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x39008a) [0x7fe75a3ed08a]\n",
      "[bt] (1) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x3906c1) [0x7fe75a3ed6c1]\n",
      "[bt] (2) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x31aba9d) [0x7fe75d208a9d]\n",
      "[bt] (3) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x31b3e58) [0x7fe75d210e58]\n",
      "[bt] (4) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x2a808f6) [0x7fe75cadd8f6]\n",
      "[bt] (5) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(MXNDArrayCreateEx+0x158) [0x7fe75cade168]\n",
      "[bt] (6) /usr/lib/x86_64-linux-gnu/libffi.so.6(ffi_call_unix64+0x4c) [0x7fe7bee55dae]\n",
      "[bt] (7) /usr/lib/x86_64-linux-gnu/libffi.so.6(ffi_call+0x22f) [0x7fe7bee5571f]\n",
      "[bt] (8) /usr/lib/python3.6/lib-dynload/_ctypes.cpython-36m-x86_64-linux-gnu.so(_ctypes_callproc+0x2b4) [0x7fe7bf069524]\n",
      "[bt] (9) /usr/lib/python3.6/lib-dynload/_ctypes.cpython-36m-x86_64-linux-gnu.so(+0x11b93) [0x7fe7bf069b93]\n",
      "\n",
      "\n",
      "Process Process-7:\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/usr/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/gluon/data/dataloader.py\", line 184, in worker_loop\n",
      "    batch = batchify_fn([dataset[i] for i in samples])\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/gluon/data/dataloader.py\", line 184, in <listcomp>\n",
      "    batch = batchify_fn([dataset[i] for i in samples])\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/data_loader.py\", line 104, in __getitem__\n",
      "    video_frame = self.transform(video_frame)\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/utils.py\", line 128, in __call__\n",
      "    img = t(img)\n",
      "  File \"/home/jiaming/code/Video-Caption-CV-NLP/utils.py\", line 93, in __call__\n",
      "    img = mx.nd.array(np.array(img).transpose(0, 3, 1, 2).astype('float32'), ctx=self.ctx)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/utils.py\", line 146, in array\n",
      "    return _array(source_array, ctx=ctx, dtype=dtype)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 2434, in array\n",
      "    arr = empty(source_array.shape, ctx, dtype)\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 3823, in empty\n",
      "    return NDArray(handle=_new_alloc_handle(shape, ctx, False, dtype))\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py\", line 139, in _new_alloc_handle\n",
      "    ctypes.byref(hdl)))\n",
      "  File \"/usr/local/lib/python3.6/dist-packages/mxnet/base.py\", line 251, in check_call\n",
      "    raise MXNetError(py_str(_LIB.MXGetLastError()))\n",
      "mxnet.base.MXNetError: [22:12:04] src/storage/storage.cc:65: Check failed: e == cudaSuccess || e == cudaErrorCudartUnloading CUDA: initialization error\n",
      "\n",
      "Stack trace returned 10 entries:\n",
      "[bt] (0) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x39008a) [0x7fe75a3ed08a]\n",
      "[bt] (1) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x3906c1) [0x7fe75a3ed6c1]\n",
      "[bt] (2) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x31aba9d) [0x7fe75d208a9d]\n",
      "[bt] (3) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x31b3e58) [0x7fe75d210e58]\n",
      "[bt] (4) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(+0x2a808f6) [0x7fe75cadd8f6]\n",
      "[bt] (5) /usr/local/lib/python3.6/dist-packages/mxnet/libmxnet.so(MXNDArrayCreateEx+0x158) [0x7fe75cade168]\n",
      "[bt] (6) /usr/lib/x86_64-linux-gnu/libffi.so.6(ffi_call_unix64+0x4c) [0x7fe7bee55dae]\n",
      "[bt] (7) /usr/lib/x86_64-linux-gnu/libffi.so.6(ffi_call+0x22f) [0x7fe7bee5571f]\n",
      "[bt] (8) /usr/lib/python3.6/lib-dynload/_ctypes.cpython-36m-x86_64-linux-gnu.so(_ctypes_callproc+0x2b4) [0x7fe7bf069524]\n",
      "[bt] (9) /usr/lib/python3.6/lib-dynload/_ctypes.cpython-36m-x86_64-linux-gnu.so(+0x11b93) [0x7fe7bf069b93]\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "if __name__ == \"__main__\":\n",
    "    main()  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
